internal/resolution/variable_sources/bundlesanddependencies/bundles_and_dependencies.go

```diff
@@ -1,4 +1,4 @@
-package bundles_and_dependencies
```

Comment:
nit: can we rename this package at all? `bundlesanddependencies` seems unnecessarily long, `bundleentity` or `bundlesource` might do the job just as well

Reply:
There is a relevant discussion on this here: https://github.com/operator-framework/operator-controller/pull/264#discussion_r1232198061

I'll look into better structure and names in a separate PR as this one is quite large already.